src/content/blog/astro-4-beta.mdx

```diff
@@ -0,0 +1,62 @@
+---
+title: "Astro 4.0 Beta"
+description: "Astro 4.0 beta is released."
+publishDate: "November 22, 2023"
+authors:
+  - matthew
+coverImage: "/src/content/blog/_images/astro-350/post-header-3.5.webp"
+socialImage: "/src/content/blog/_images/astro-350/og-image-3.5.webp"
+lang: "en"
+---
+
+Today we've released the first beta of Astro 4! Those feeling adventurous can install new beta, review our in-progress migration docs, and report any bugs you find to the [Astro discord](https://astro.build/chat).
+
+## What to expect in Astro 4
+
+We're releasing Astro 4 stay up to date with the latest version of Vite, with Vite 5 having just been released last month.
+
+As Vite 5 is *also* a small major, most projects will be able to upgrade to Astro 4 with only minimal changes to your code, if any at all.
+
+Additionally with Astro 4 we are planning to bring some of the features we have been working on over the last few months out of experimental status and mark them as stable. We are still deciding *which* features those will be. Stay tuned for more!
+
+## Trying out the beta
+
+You can try out the new beta in your existing app by installing the `astro@beta` package with your favorite package manager.
+
+```sh
+npm install astro@beta
+pnpm install astro@beta
+yarn install astro@beta
+```
+
+If starting a new project you can also try out the beta by running:
+
+```sh
+npm init astro@beta
+```
+
+Again, substitute with your favorite package manager. We have not completed the migration guide yet, but you can see what is being changed in the [pull request] (LINK NEEDED)
+
+## Astro 3.6
+
+Additionally we've recently released Astro 3.6 which comes with new events in the `<ViewTransitions />` router as well as fallback for `prefetch` in reduced data scenarios.
+
+### View Transition events
+
+Astro 3.6 adds these three new events as part of the `<ViewTransitions />` router's lifecycle:
+
+`astro:before-preparation` runs immediately before the *preparation* step; where the content of the new page is loaded. This can be used to override the behavior of loading the new page, for example to show a loading spinner.
+
+`astro:after-preparation` runs after the content has been loaded. This can be used to *stop* a loading spinner, if one was added as part of `astro:before-preparation`.
+
+`astro:before-swap` runs immediately before the *swap* step; where the contents of the old page are swapped out in favor of the contents of the new page. This event includes the `event.newDocument` property, a `Document` that you can mutate. One use-case for this event is to set dark mode class names if using `localStorage` to store light/dark mode preferences.
```

Comment:
```suggestion
`astro:before-preparation` runs immediately before the *preparation* step, where the content of the new page is loaded. This can be used to override the behavior of loading the new page, for example to show a loading spinner.

`astro:after-preparation` runs after the content has been loaded. This can be used to *stop* a loading spinner if one was added as part of `astro:before-preparation`.

`astro:before-swap` runs immediately before the *swap* step, where the contents of the old page are swapped out in favor of the contents of the new page. This event includes the `event.newDocument` property, a `Document` that you can mutate. One use-case for this event is to set dark mode class names if using `localStorage` to store light/dark mode preferences.
```
grammar nits only!